posts/2020-03-31-Rust-survey-2019.md

```diff
@@ -0,0 +1,216 @@
+
+## Rust Adoption - a Closer Look
+
+First, we asked what would make Rust more appealing to your company and convince them to adopt it.
+
+![How can we make Rust more appealing to your company](images/2020-03-RustSurvey/21-Rust-more-appealing.png)
+
+Several users gave specific examples:
+* "Smoothest learning curve as possible, as a small business even 4-6 weeks to become productive is a lot to ask"
+* "Higher market penetration"
+* "More stable libraries"
+* "A full-stack web framework like Rails, Django and Phoenix"
+* "Better documentation, more examples, recommendation on what crates to use"
+* "More emphasis on how it is a safer alternative to C or C++ (and really should be the default usually).”
+* "Improve compile times. Compiling development builds at least as fast as Go would be table stakes for us to consider Rust. (Release builds can be slow."
+* "Better platform support"
+* "Security and performance, cost efficient and "green" (low carbon footprint) language"
+* "Embedded development targeting ARM"
+* "Better GUI framework, similar to Qt or directly using Qt via bindings."
+
+Most indicated that Rust maturity - such as more libraries and complete learning resources and more mature production capabilities - would make Rust more appealing.
+
+Let's take a closer look at each of these, starting with the need for more mature libraries.
+
+## Libraries - a Closer Look
+
+When we asked users what libraries they consider critical to the Rust ecosystem, these were the top ten responses:
+* serde
+* rand
+* tokio
+* async
+* clap
+* regex
+* log
+* futures
+* hyper
+* lazy_static
+
+We also asked how many dependencies users were using were 1.0 or above.
+* 0.8% indicated "All"
+* 6.7% indicated "Most"
+* 65.9% indicated "Some"
+* 5.2% indicated "None"
+* 21.4% indicated "I don't know"
+
+[Add COPY/analysis here]
+
+## IDEs and Tooling - a Closer Look
+
+IDE support for Rust was also cited as a barrier to adoption.
+
+When we asked users what editors they use, Vim and VSCode were the most popular by far, followed by Intellij.
+
+We also asked what IDE setups users used:
+* 43.3% indicated RLS
+* 21.7% indicated Intellij
+* 15.2% indicated Rust-analyzer
+* 12.4% indicated No (or CTAGS)
+* 4.2% indicated Only Racer
+
+As for platforms that users develop on - Linux and Windows continue to dominate. 
+* 55% of Rust users develop on Linux
+* 24% develop on Windows
+* 23% develop on macOS
+
+![Which Rust version do you use for your applications?](images/2020-03-RustSurvey/16-Which-Rust-Version.png)
+
+The vast majority of users use the current stable version of Rust (63%). Surprisingly, the number of users using the Nightly compiler in their workflow is down at 20%. Last year it was at over 56%.
+
+[ADD COPY/analysis]
+
+## Learning Curve - a Closer Look
+
+Rust is well known for its significant learning curve. 
+
+![How would you rate your Rust expertise](/images/2020-03-RustSurvey/13-Rust-Expertise-and-Time-In-Rust.png)
```

Comment:
It loosely seems like the width of the bars in this graph is changing as we go to the right? Maybe it's my eyes...